feature/security/gnsi/credentialz/tests/password_console_login/password_console_login_test.go

```diff
@@ -0,0 +1,310 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package password_console_login
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/openconfig/featureprofiles/internal/deviations"
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	gpb "github.com/openconfig/gnmi/proto/gnmi"
+	"github.com/openconfig/gnsi/credentialz"
+	tpb "github.com/openconfig/kne/proto/topo"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/binding"
+	"github.com/openconfig/ondatra/gnmi"
+	"github.com/openconfig/ondatra/gnmi/oc"
+	"golang.org/x/crypto/ssh"
+)
+
+const (
+	username          = "testuser"
+	password          = "i$V5^6IhD*tZ#eg1G@v3xdVZrQwj"
+	passwordVersion   = "v1.0"
+	passwordCreatedOn = 1705962293
+	// think this should work for all vendors contributing to fp
+	command = "show version"
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+func createNativeRole(t testing.TB, dut *ondatra.DUTDevice, role string) {
+	t.Helper()
+	switch dut.Vendor() {
+	case ondatra.NOKIA:
+		roleData, err := json.Marshal([]any{
+			map[string]any{
+				"services": []string{"cli", "gnmi"},
+			},
+		})
+		if err != nil {
+			t.Fatalf("Error with json Marshal: %v", err)
+		}
+
+		userData, err := json.Marshal([]any{
+			map[string]any{
+				"role": []string{role},
+			},
+		})
+		if err != nil {
+			t.Fatalf("Error with json Marshal: %v", err)
+		}
+
+		SetRequest := &gpb.SetRequest{
+			Prefix: &gpb.Path{
+				Origin: "native",
+			},
+			Replace: []*gpb.Update{
+				{
+					Path: &gpb.Path{
+						Elem: []*gpb.PathElem{
+							{Name: "system"},
+							{Name: "aaa"},
+							{Name: "authorization"},
+							{Name: "role", Key: map[string]string{"rolename": role}},
+						},
+					},
+					Val: &gpb.TypedValue{
+						Value: &gpb.TypedValue_JsonIetfVal{
+							JsonIetfVal: roleData,
+						},
+					},
+				},
+				{
+					Path: &gpb.Path{
+						Elem: []*gpb.PathElem{
+							{Name: "system"},
+							{Name: "aaa"},
+							{Name: "authentication"},
+							{Name: "user", Key: map[string]string{"username": username}},
+						},
+					},
+					Val: &gpb.TypedValue{
+						Value: &gpb.TypedValue_JsonIetfVal{
+							JsonIetfVal: userData,
+						},
+					},
+				},
+			},
+		}
+		gnmiClient := dut.RawAPIs().GNMI(t)
+		if _, err := gnmiClient.Set(context.Background(), SetRequest); err != nil {
+			t.Fatalf("Unexpected error configuring User: %v", err)
+		}
+	default:
+		t.Fatalf("Unsupported vendor %s for deviation 'deviation_native_users'", dut.Vendor())
+	}
+}
+
+func setupUser(t *testing.T, dut *ondatra.DUTDevice) {
+	auth := &oc.System_Aaa_Authentication{}
+	auth.GetOrCreateUser(username)
+
+	gnmi.Update(t, dut, gnmi.OC().System().Aaa().Authentication().Config(), auth)
```

Comment:
I think you can do this all with gnsi.credentialz itself.

Reply:
It seems we cannot set system usernames through gnsi.credentialz. As per proto definition [here](https://github.com/openconfig/gnsi/blob/cb2461425cff43ff10b5109ba264fcd71008630e/credentialz/credentialz.proto#L498-L499), an account should already exist.

Reply:
That message is for SshAuthorizedPrincipal, which yes that'd need to already exist. We shouldn't be using AuthorizedPrincipalFile for hiba tests. HIBA is an implementation of AuthorizedPrincipalsCommand.

Reply:
I thought that'd do the trick right?

https://github.com/openconfig/featureprofiles/tree/c07c43705ff0a8eb6180607709fe3d67f281dd73/feature/security/gnsi/credentialz/tests#configure-a-testuser-and-password

Reply:
The Account message also requires user to already exist per the [proto](https://github.com/openconfig/gnsi/blob/cb2461425cff43ff10b5109ba264fcd71008630e/credentialz/credentialz.proto#L552-L553). The Readme needs to be revisited for this test.

Reply:
That doesn't sound right to me, requring a username setup in gnmi set first, then allowing account setting in gnsi.credentialz. We should change that comment and clarify the behavior. I can talk to Tomek about why that comment says that.

Reply:
Thanks, will wait for the clarification.